sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -733,7 +734,7 @@ def connect(self):
             ) from None
         try:
             self.sock = create_connection(
-                url="wss://{}".format(self._custom_endpoint or self._host),
+                url="wss://{}".format(self._custom_endpoint or self._host) if self._use_tls else "ws://{}".format(self._custom_endpoint) or self._host,
```

Comment:
if you see errors still popping up, you will have to modify how the ssl context object is set up

`ctx.check_hostname = False`
`ctx.verify_mode = ssl.CERT_NONE`